packages/std/src/coins.rs

```diff
@@ -31,13 +30,126 @@ pub fn coin(amount: u128, denom: &str) -> Coin {
     Coin::new(amount, denom)
 }
 
-/// has_coins returns true if the list of coins has at least the required amount
-pub fn has_coins(coins: &[Coin], required: &Coin) -> bool {
-    coins
-        .iter()
-        .find(|c| c.denom == required.denom)
-        .map(|m| m.amount >= required.amount)
-        .unwrap_or(false)
+// Wallet wraps Vec<Coin> and provides some nice helpers. It mutates the Vec and can be
+// unwrapped when done.
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, JsonSchema)]
+pub struct Wallet(pub Vec<Coin>);
+
+impl Wallet {
+    pub fn into_vec(self) -> Vec<Coin> {
+        self.0
+    }
+
+    /// returns true if the list of coins has at least the required amount
+    pub fn has(&self, required: &Coin) -> bool {
+        self.0
+            .iter()
+            .find(|c| c.denom == required.denom)
+            .map(|m| m.amount >= required.amount)
+            .unwrap_or(false)
+    }
+
+    /// normalize Wallet (sorted by denom, no 0 elements, no duplicate denoms)
+    pub fn normalize(&mut self) {
```

Comment:
Shouldn't the data ba stored in a map internally, making it impossible to have normalized an unnormalized instances?

`Wallet::from_vec()` could read them from a vector into a map.

Reply:
Seems overhead, but could do it